x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/ScriptFieldMapper.java

```diff
@@ -112,13 +130,13 @@ protected Builder(String name, ScriptCompiler scriptCompiler) {
 
         @Override
         public ScriptFieldMapper build(BuilderContext context) {
-            MappedFieldType mappedFieldType;
-            if (runtimeType.getValue().equals("keyword")) {
-                StringScriptFieldScript.Factory factory = scriptCompiler.compile(script.getValue(), StringScriptFieldScript.CONTEXT);
-                mappedFieldType = new RuntimeKeywordMappedFieldType(buildFullName(context), script.getValue(), factory, meta.getValue());
-            } else {
+            BiFunction<Builder, BuilderContext, MappedFieldType> fieldTypeResolver = Builder.FIELD_TYPE_RESOLVER.get(
```

Comment:
@nik9000 you may have opinions on this. I added it because I did not want to have tests to maintain a list of supported runtime types and rather rely on the truth which cannot get outdated. Downside is we have no if but rather a map with a bifunction, maybe a bit cryptic.

Reply:
I'm pretty used to maps to functions at this point. I'm happy with this. I'm not 100% this is better than the `switch` statement that I had in #59721, but it does the job just as well. If you like it, I'm happy with it.

Reply:
it does not do a better job than the switch, but it allows to expose all the keys for testing :)